src/Microsoft.PowerShell.PSReadLine/ConsoleLib.cs

```diff
@@ -1068,17 +1068,33 @@ public void WriteBufferLines(BufferChar[] buffer, ref int top, bool ensureBottom
 
             Console.SetCursorPosition(0, (top>=0) ? top : 0);
 
+            var lastForegroundColor = (ConsoleColor)(-1);
+            var lastBackgroundColor = (ConsoleColor)(-1);
             for (int i = 0; i < buffer.Length; ++i)
             {
                 // TODO: use escape sequences for better perf
-                Console.ForegroundColor =  buffer[i].ForegroundColor;
-                Console.BackgroundColor =  buffer[i].BackgroundColor;
+                var nextForegroundColor = buffer[i].ForegroundColor;
+                var nextBackgroundColor = buffer[i].BackgroundColor;
+                if (lastForegroundColor != nextForegroundColor)
```

Comment:
Small nitpick, but it seems that `if (nextForegroundColor != lastForegroundColor)` is slightly more correct improving readability